clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -104,6 +109,9 @@ public void handleMessage(EntityResponse messageFromServer) {
 
       @Override
       public byte[] createExtendedReconnectData() {
+        for (ReconnectListener reconnectListener : reconnectListeners) {
```

Comment:
Since multiple threads can create different caches concurrently, `CopyOnWriteArrayList` just helps there. Anyways its just the number of caches, so should not be a problem